static/js/components/RepeatableContentListing.tsx

```diff
@@ -71,43 +72,42 @@ export default function RepeatableContentListing(props: {
     event: ReactMouseEvent<HTMLLIElement | HTMLAnchorElement, MouseEvent>
   ) => {
     event.preventDefault()
-    setPanelState({
-      textId,
-      formType: textId ? ContentFormType.Edit : ContentFormType.Add
-    })
+
+    setDrawerState(
+      textId ? createModalState("editing", textId) : createModalState("adding")
```

Comment:
Can we make enums of `adding`, `editing`, and `closed`?

Reply:
I considered doing so but decided not to, this was my reasoning:

1. It makes calls to the `createModalState` function longer because we'd have to reference the enum member (i.e. `createModalState("closed")` vs `createModalState(ModalStateVariant.Closed)`)
2. it adds another import, so that instead of just importing `createModalState` we'd need to import the enum as well
3. it doesn't provide any additional type safety beyond what we get from the function overloading. If you try doing something like `createModalState("dancing")` you'll get a type error saying `"No overload matches this call"`

so that made me think it wasn't actually _so_ helpful to create an enum for these values as opposed to just using these little string constants. what do you think?